sov-modules/sov-modules-impl/bank/src/call.rs

```diff
@@ -1,56 +1,102 @@
-use anyhow::Result;
+use crate::{Amount, Bank, Coins, Token};
+use anyhow::{bail, Result};
 use sov_modules_api::CallResponse;
 use sov_state::WorkingSet;
 
-use crate::{Amount, Bank, Coins};
-
+/// This enumeration represents the available call messages for interacting with the bank module.
 #[derive(borsh::BorshDeserialize, borsh::BorshSerialize, Debug, PartialEq)]
 pub enum CallMessage<C: sov_modules_api::Context> {
+    /// Creates a new token with the specified name and initial balance.
     CreateToken {
+        /// salt: a random value use to create a unique token address.
+        salt: u64,
+        /// token_name: the name of the new token.
         token_name: String,
+        /// initial_balance: the initial balance of the new token.
         initial_balance: Amount,
+        /// minter_address: the address of the account that minted new tokens.
         minter_address: C::Address,
     },
 
+    /// Transfers a specified amount of tokens to the specified address.
     Transfer {
+        /// to: the address to which the tokens will be transferred.
         to: C::Address,
+        /// coins: the amount of tokens to transfer.
         coins: Coins<C::Address>,
     },
 
+    /// Burns a specified amount of tokens.
     Burn {
+        /// coins: the amount of tokens to burn.
         coins: Coins<C::Address>,
     },
 }
 
 impl<C: sov_modules_api::Context> Bank<C> {
-    pub fn create_token(
+    pub(crate) fn create_token(
         &self,
-        _token_name: String,
-        _initial_balance: Amount,
-        _minter_address: C::Address,
-        _context: &C,
-        _working_set: &mut WorkingSet<C::Storage>,
+        token_name: String,
+        salt: u64,
+        initial_balance: Amount,
+        minter_address: C::Address,
+        context: &C,
+        working_set: &mut WorkingSet<C::Storage>,
     ) -> Result<CallResponse> {
-        // This function will create a unique address for `token_name` and insert, the new `Token` to self.tokens
-        todo!()
+        let token_address = super::create_token_address::<C>(&token_name, context.sender(), salt);
+
+        match self.tokens.get(&token_address, working_set) {
```

Comment:
nit: It think it would be slightly more idiomatic not to `match` here. 

```
if self.tokens.get(&token_address, working_set).is_some() {
  bail!("Token address already exists");
}
...
```

Reply:
done